packages/js/src/plugins/candyMachineModule/operations/findCandyMachinesByPublicKeyField.ts

```diff
@@ -121,8 +122,45 @@ export const findCandyMachinesByPublicKeyFieldOperationHandler: OperationHandler
             ? parseCandyMachineCollectionAccount(unparsedCollectionAccount)
             : null;
 
-          return toCandyMachine(account, unparsedAccount, collectionAccount);
+          return {
+            account,
+            unparsedAccount,
+            collectionAccount,
+          };
         }
       );
+
+      // Find mint details for all unique SPL tokens used
+      // in candy machines that have non-null `tokenMint`
+      const tokenMints = new Set(
+        candyMachinesDetails
+          .map((details) => details.account.data.tokenMint)
+          .filter((tokenMint): tokenMint is PublicKey => tokenMint !== null)
+      );
+
+      let mints: Mint[] = [];
+
+      if (tokenMints.size > 0) {
+        const unparsedMintAccounts = await metaplex
+          .rpc()
+          .getMultipleAccounts([...tokenMints], commitment);
```

Comment:
If you're up for the challenge, we could combine this `getMultipleAccounts` call with the one that fetches collections since they can both be done in parallel. We'd just need to slice the returned accounts so we know which ones are collections and which ones are mint.

Reply:
Sure!